### PR TITLE
Portal refactor & `Anchor` implementation

### PR DESCRIPTION
**Pull request by creativecreatorormaybenot**
## Description

* Implements passing the root bounds to portal anchors, i.e. passing the `portalRect` (renamed from `overlayRect`) to `getFollowerOffset` (renamed from `getSourceOffset`).
* Implements handling of off-screen cut-off with a dedicated example.
* Refactors the library to clean up all concepts to use only `Portal`, `PortalTarget`, `PortalFollower`, and `Anchor`.
* Updates all documentation and examples.

### Implementation details

The passing of root bounds is achieved using a custom `FollowerLayer` implementation that takes a callback for computing the `linkedOffset`. This allows making the `linkedOffset` depend on the offset of the leader layer (via `CompositedTransformTarget`). And this in turn allows to make the `linkedOffset` depend on the `theaterRect`, which requires only the relative top-left offset to the `targetRect` and the size of the theater, which we can obtain via the `_RenderPortalTheater` render box.

## Tests

This PR only contains two unit test cases, but I did not feel the need to add more test cases since these are already the cases that cover the "most possible" edge cases, i.e. a portal that is not the root of the app and portal entry sizes that differ from the portal and child sizes.

### Fragility

Doing this with TDD **really** helped me to come to a **non-fragile** implementation. Initially, I thought it would be enough to call `localToGlobal` with `Offset.zero` and the theater render box ancestor. However, this does not work because it does not respect the offset of the `LayerLink`. After that, I thought that I could access the `link.leader` in `paint` directly. Using a non-null assertion worked fine in my example app case, however, the unit tests revealed that this was actually **not** safe to do on the first paint call. This is because the layer link leader only attaches itself **after** the painting phase. And this is how I finally arrived at creating a custom `FollowerLayer`, which allows to access the leader layer offset in the **composition phase**.

This is why I believe that the new implementation is **non-fragile** and solid.

We can of course add more test cases - it cannot hurt :)

### CODE REVIEW

**Comment by creativecreatorormaybenot** [comment]
Current state: the new concepts are introduced and applied everywhere. I have touched all of the docs and examples to use the new concepts and it all seems coherent to me.

What is still open is handling the off-screen cut-off.

File: README.md [hunk 1]
```diff
@@ -247,6 +247,57 @@ Center(
...
+The portal also defines the area (rectangle bounds) that are available to any
+followers to be rendered onto the screen.
+
+In detail, you might wrap your whole `MaterialApp` in a single `Portal` widget,
+which would mean that you can use the whole area of your app to render followers
+attached to targets that are children of the `Portal` widget.
+
+### Target
+
+A target is any place within a portal that can be followed by a follower. This
```
Comment (venkatd) [inline README.md:275]: @creativecreatorormaybenot perhaps we can add 1-2 examples somewhere? For example an autocomplete dropdown and a textfield and a tooltip and an avatar. Both of these are examples of follower=>target.

**Comment by creativecreatorormaybenot** [comment]
Closing this here as the changes have been merged upstream in https://github.com/fzyzcjy/flutter_portal/pull/44.
